server.go

```diff
@@ -152,19 +155,22 @@ func (server *Server) acceptLoop(listener *Listener) error {
 			return err
 		}
 
-		server.accepts.Inc(1)
+		// for cases of probe or blackhole connection
+		if err := conn.SetDeadline(time.Now().Add(server.Timeout)); err != nil {
+			continue
+		}
 
-		if server.MaxConns > 0 && server.numConns.Count() >= server.MaxConns {
-			// too many connections
+		if server.MaxConns > 0 && server.numConns.Count() > server.MaxConns {
 			server.tooManyConns.Inc(1)
-			conn.Close()
+			_ = conn.Close()
 			continue
 		}
 
+		server.accepts.Inc(1)
```

Comment:
shouldn't this happen directly after `listener.Accept ` returned without an error? As it count the number of accepts before we close connection incase of too many open connections?